src/algebra/order_functions.lean

```diff
@@ -64,6 +64,17 @@ begin
   exact ⟨h₁ h.1, λ e, h.2 (h₂ e)⟩
 end
 
+lemma nat.strict_mono_of_lt_succ {α : Type*} [preorder α] (f : ℕ → α)
+  (hf : ∀ n, f n < f n.succ) : strict_mono f :=
+begin
+  assume a b hab,
+  cases nat.exists_eq_add_of_lt hab with k hk,
```

Comment:
inducting on `hab` should shorten the proof.

Reply:
Here is a one-liner: https://github.com/fpvandoorn/mathlib/blob/cubes/src/cube.lean#L21